### PR TITLE
Add explanatory comment to annotation box example

### PR DESCRIPTION
**Pull request by mollyrossow**
## PR Summary
This pull request adds explanatory text to the Demo Annotation Box gallery example in response to issue #11654.

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/text_labels_and_annotations/demo_annotation_box.py [hunk 1]
```diff
@@ -3,7 +3,12 @@
 Demo Annotation Box
 ===================
 
+The AnnotationBbox Artist creates an annotation on some location using an
```
Comment (jklymak) [inline examples/text_labels_and_annotations/demo_annotation_box.py]: suggest strike: "on some location".  Vague and doesn't help the sentence....

File: examples/text_labels_and_annotations/demo_annotation_box.py [hunk 2]
```diff
@@ -3,7 +3,12 @@
...
+OffsetBox. This example demonstrates three different OffsetBoxes: TextArea,
+DrawingArea and OffsetImage. AnnotationBbox gives more fine-grained control
+than using the axes method annotate.
```
Comment (jklymak) [inline examples/text_labels_and_annotations/demo_annotation_box.py]: Can we link as `.axes.Axes.annotate`?

File: examples/text_labels_and_annotations/demo_annotation_box.py (hunk 1 above)
Comment (mollyrossow) [reply examples/text_labels_and_annotations/demo_annotation_box.py]: Fixed.

**Comment by jklymak** [comment]
Thanks a lot @mollyrossow !

File: examples/text_labels_and_annotations/demo_annotation_box.py (hunk 2 above)
Comment (mollyrossow) [reply examples/text_labels_and_annotations/demo_annotation_box.py]: Link added.

**Pull request merged**